# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [6]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [7]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5         V6        V7       
1 0.9334692 0.1730888 0.5866236 0.9682807 0.17827303 0.6755758 0.4883920
2 0.5300901 0.3734534 0.4567416 0.1449273 0.01308794 0.4761931 0.2095939
  V8        V9         V10       V11       V12       V13       V14      
1 0.4956691 0.13011577 0.9418487 0.8181827 0.8114426 0.6379464 0.7572795
2 0.7605389 0.08938114 0.1221663 0.1371139 0.5452560 0.1918477 0.4196751
  V15       V16       V17       V18       V19         V20      
1 0.5736074 0.7540943 0.1056243 0.2586289 0.288887934 0.4946214
2 0.4062597 0.2010069 0.5110908 0.8412645 0.008263659 0.2621657

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [8]:
df2 <- df1 %>%
  

head(df2)

ERROR: Error in stopifnot(length(n) == 1L): object 'df2' not found


## Assignment 2

We will generate some data:

In [9]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [10]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [11]:
df5 <- merge(df3, df4, all='TRUE')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook